python/cudf/cudf/io/csv.py

```diff
@@ -127,6 +136,34 @@ def to_csv(
                 "Dataframe doesn't have the labels provided in columns"
             )
 
+    if sep == "-":
+        # TODO: Remove this error once following issue is fixed:
+        # https://github.com/rapidsai/cudf/issues/6699
+        if any(
+            isinstance(col, cudf.core.column.DatetimeColumn)
+            for col in df._data.columns
+        ):
+            raise ValueError(
```

Comment:
Would it be better to have test cases to these exceptions ?

Reply:
Yep, done 👍